### DISCUSSION TITLE
Set DataGrid vertical scrollbar to DataGrid height or scrollbar track height

### DISCUSSION THREAD

**Discussion by TangDave** [AvaloniaUI/Avalonia#15979] [Q&A] [answered]
Currently, I have the following in my App.axaml file to give a more reasonable 'large change' increment to the DataGrid vertical scrollbar - this was provided in my previous question at https://github.com/AvaloniaUI/Avalonia/discussions/15968

```
		<Style Selector="DataGrid /template/ ScrollBar#PART_VerticalScrollbar">
			<Setter Property="LargeChange" Value="120"/>
		</Style>
```

However, it would be ideal to have the value be set to either the DataGrid height or the vertical scrollbar track height. 
Is this possible?

**Comment by stevemonaco** [comment, marked as answer]
Try `Value="{Binding $self.Bounds.Height}"` or `Value="{Binding $parent[DataGrid].Bounds.Height}"`.

**Reply by TangDave** [reply]
Exactly what I was looking for - you'd think this would be the obvious default setting for paging with the vertical scrollbar.

**Reply by stevemonaco** [reply]
Not at all, but I'm not sure what the WPF default was either. There are several paging strategies: distance in pixels, number of rows, percentage of viewport, etc. Some will prefer half screen scrolls. XAML frameworks have a harder time in this area than CSS which has richer ways to specify these sort of properties. `Grid` is one of few controls that has more complex specifications and had to introduce its own type (`GridLength`) with custom handling to sort out pixel, proportional, and automatic sizing properties.
